### PR TITLE
Blog post on the deprecation of the Istio sidecar injection annotation

### PR DESCRIPTION
**Pull request by nataliasitko**
Changes proposed in this pull request:

- Initial blog post on the deprecation of the Istio sidecar injection annotation

**Related issue(s)**
- https://github.com/kyma-project/website/issues/932
- https://github.com/kyma-project/kyma/issues/16374

### CODE REVIEW

File: content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+## Verify which Pods use the sidecar injection annotation
+
+To see which Pods in your workloads have the `sidecar.istio.io/inject` annotation set, run the following kubectl command:
+```
+Insert a kubectl command
+```
+Or the following istioctl command:
+```
+Insert a istioctl command
```
Comment (strekm) [inline content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md:27]:
```suggestion
istioctl analyze -n {NAMESPACE}
```

File: content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+```
+
+Here's an example of a Pod with the `sidecar.istio.io/inject` annotation set to `true`:
+```
+Insert an example
```
Comment (strekm) [inline content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md:32]:
```suggestion
apiVersion: v1
kind: Pod
metadata:
  name: example-workload
  namespace: test-namespace
  annotations:
    sidecar.istio.io/inject: "true"
spec:
  containers:
  - name: istio-proxy
    image: eu.gcr.io/kyma-project/external/istio/proxyv2:1.16.1-distroless
    ...
  - image: docker.io/kennethreitz/httpbin
    name: example-workload
  ...
```

File: content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+You must replace `sidecar.istio.io/inject` annotation with the `sidecar.istio.io/inject` label on each of the Pods listed after executing the previous commands. In order to do so, ...
+
+```
+Insert instructions
+```
+
+Here's an example of a Pod that has the Istio sidecar injection annotation successfully replaced with the label:
+```
+Insert an example
```
Comment (strekm) [inline content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md:45]:
```suggestion
apiVersion: v1
kind: Pod
metadata:
  name: example-workload
  namespace: test-namespace
  labels:
    sidecar.istio.io/inject: "true"
spec:
  containers:
  - name: istio-proxy
    image: eu.gcr.io/kyma-project/external/istio/proxyv2:1.16.1-distroless
    ...
  - image: docker.io/kennethreitz/httpbin
    name: example-workload
  ...
```

File: content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+## Overview
+To enable injecting the Istio sidecar proxy into a Pod, you can label the entire Namespace the Pod belongs to or the Pod itself. For more information about the Istio sidecar proxy injection, read the [Istio documentation](https://istio.io/latest/docs/setup/additional-setup/sidecar-injection/) and visit [this blog post](https://kyma-project.io/docs/kyma/main/04-operation-guides/operations/smsh-01-istio-enable-sidecar-injection/).
+
+So far, you could modify a workloads's configuration using either the `sidecar.istio.io/inject` annotation or the `sidecar.istio.io/inject` label. From version 1.16 of Istio, **using the annotation is no longer fully supported**. Therefore, you must verify which of your Pods have the Istio sidecar injection annotation defined and replace each of these annotations with the `sidecar.istio.io/inject` label.
```
Comment (strekm) [inline content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md:17]:
```suggestion
So far, you could modify a workloads's configuration using either the `sidecar.istio.io/inject` annotation or the `sidecar.istio.io/inject` label. From version 1.16 of Istio, **the annotation is deprecated** in favour of the label. Therefore, you must verify which of your Pods have the Istio sidecar injection annotation defined and replace each of these annotations with the `sidecar.istio.io/inject` label.
```

File: content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+## Verify which Pods use the sidecar injection annotation
+
+To see which Pods in your workloads have the `sidecar.istio.io/inject` annotation set, run the following kubectl command:
```
Comment (strekm) [inline content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md:21]:
```suggestion
To see which Pods in your namespace have the `sidecar.istio.io/inject` annotation set, run the following kubectl command:
```

File: content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    image: eu.gcr.io/kyma-project/external/istio/proxyv2:1.16.1-distroless
+    ...
+  - image: docker.io/kennethreitz/httpbin
+    name: example-workload
+  ...
+```
+
+## Change the sidecar injection annotations to labels
+
+You must replace `sidecar.istio.io/inject` annotation with the `sidecar.istio.io/inject` label on each of the Pods listed after executing the previous commands. In order to do so, ...
```
Comment (strekm) [inline content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md:51]:
```suggestion
You must replace `sidecar.istio.io/inject` annotation with the `sidecar.istio.io/inject` label on each of the Pods listed after executing the previous commands. In order to do so, you must replace annotation in favour of label in your workload's Pod's template.
```

**Review by strekm — APPROVED** [review]
/meow

File: content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+## Overview
+To enable injecting the Istio sidecar proxy into a Pod, you can label the entire Namespace the Pod belongs to or the Pod itself. For more information about the Istio sidecar proxy injection, read the [Istio documentation](https://istio.io/latest/docs/setup/additional-setup/sidecar-injection/) and visit [this blog post](https://kyma-project.io/docs/kyma/main/04-operation-guides/operations/smsh-01-istio-enable-sidecar-injection/).
+
+So far, you could modify a workload's configuration using either the `sidecar.istio.io/inject` annotation or the `sidecar.istio.io/inject` label. With the 1.16 release of Istio, **the annotation was deprecated** in favor of the label. Therefore, you must verify which of your Pods have the Istio sidecar injection annotation set and replace each of these annotations with the `sidecar.istio.io/inject` label. **Istio will get upgraded to version 1.16 with the release of Kyma 2.10**, but you can start the necessary preparations already.
```
Comment (IwonaLanger) [inline content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md:17]:
```suggestion
So far, you could modify a workload's configuration using either the `sidecar.istio.io/inject` annotation or the `sidecar.istio.io/inject` label. With the 1.16 release of Istio, **the annotation was deprecated** in favor of the label. Therefore, you must verify which of your Pods have the Istio sidecar injection annotation set and replace each of these annotations with the `sidecar.istio.io/inject` label. **Istio will get upgraded to version 1.16 with the release of Kyma 2.10**, but you can already start the necessary preparations. 
```

File: content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  - name: istio-proxy
+    image: eu.gcr.io/kyma-project/external/istio/proxyv2:1.16.1-distroless
+    ...
+  - image: docker.io/kennethreitz/httpbin
+    name: example-workload
+  ...
+```
+
+## Final remarks
+ Read [this blog post](https://kyma-project.io/docs/kyma/latest/01-overview/main-areas/service-mesh/smsh-03-istio-sidecars-in-kyma) to discover the benefits that the Istio sidecar proxy brings. In [the blog section](https://kyma-project.io/blog/) on the Kyma website, you can find all the essential news and tutorials related to Kyma Istio.
```
Comment (IwonaLanger) [inline content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md:85]:
```suggestion
 Read [this blog post](https://kyma-project.io/docs/kyma/latest/01-overview/main-areas/service-mesh/smsh-03-istio-sidecars-in-kyma) to discover the benefits that the Istio sidecar proxy brings. In [the blog](https://kyma-project.io/blog/) on the Kyma website, you can find all the essential news and tutorials related to Kyma Istio.
```

File: content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+  <details>
+    <summary label="istioctl">
+    istioctl
+    </summary>
+
+    ```
+    istioctl analyze -n {NAMESPACE}
+    ```
+  </details>
+</div>
```
Comment (IwonaLanger) [inline content/blog-posts/2022-12-21-deprecation-of-istio-sidecar-injection-annotation/index.md:41]: The first dropdown and the code in both of them do not render correctly.

**Review by IwonaLanger — APPROVED** [review]

**Review by IwonaLanger — APPROVED** [review]

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**